flocker/node/agents/blockdevice.py

```diff
@@ -1286,18 +1357,22 @@ def resize_volume(self, blockdevice_id, size):
         This implementation is limited to being able to resize volumes only if
         they are unattached.
         """
-        backing_path = self._unattached_directory.child(
-            blockdevice_id.encode("ascii")
-        )
+        volume = get_blockdevice_volume(self, blockdevice_id)
+        filename = _backing_file_name(volume)
+        backing_path = self._unattached_directory.child(filename)
+        new_size = allocated_size(self.allocation_unit(), size)
```

Comment:
Same comment as above; I don't think loopback implemetnation should do rounding itself.